lib/domain-shared/shared_use_cases/account_use_case.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:seeds/datasource/remote/firebase/firebase_message_token_repository.dart';
+
+abstract class AccountUseCase {
```

Comment:
I dont love this abstraction 

updateFirebaseToken - RemoveToken or Set Token. Neither of the actions is an update. 

I would remove this abstraction and just call removeFirebaseMessageToken or setFirebaseMessageToken as needed in the use case.

Reply:
Then we duplicate this code all over the place?

This is an important bugfix we need to merge into the code - we can't block this all the time.

Reply:
updateFirebaseToken is responsible for making sure we do the right thing whenever the account name changes. 

I had this as part of set account, which IMO maybe actually better since it ensures this bug this PR is fixing can never happen again.

Use cases are a little more error prone and less secure, but also less entangled, so I can see how they are OK to use as well. 

But in reality when I fix a bug the best outcome is that programmer error can no longer cause the bug. That's what it's aiming for.

Reply:
I'll review again in 1 hour. Having dinner